src/config.ts

```diff
@@ -224,34 +224,28 @@ export function getDefaultConfig(): IConfig {
     return Config.util.cloneDeep(defaultConfig);
 }
 
-/**
- * Grabs an explicit path provided for mjolnir's config from an arguments vector if provided, otherwise returns undefined.
- * @param argv An arguments vector sourced from `process.argv`.
- * @returns The path if one was provided or undefined.
- */
-function configPathFromArguments(argv: string[]): undefined | string {
-    const configOptionIndex = argv.findIndex(arg => arg === "--draupnir-config");
-    if (configOptionIndex > 0) {
-        const configOptionPath = argv.at(configOptionIndex + 1);
-        if (!configOptionPath) {
-            throw new Error("No path provided with option --draupnir-config");
-        }
-        return configOptionPath;
-    } else {
-        return;
-    }
-}
-
 export function read(): IConfig {
```

Comment:
```
/**
 * @returns The users's raw config, deep copied over the `defaultConfig`.
 */
function readConfigSource(): IConfig {
    const explicitConfigPath = getCommandLinePathOption(process.argv, "--draupnir-config", true);
    if (explicitConfigPath !== undefined) {
        const content = fs.readFileSync(explicitConfigPath, "utf8");
        const parsed = load(content);
        return Config.util.extendDeep({}, defaultConfig, parsed);
    } else {
        return Config.util.extendDeep({}, defaultConfig, Config.util.toObject()) as IConfig;
    }
}

export function read(): IConfig {
    const config = readConfigSource();
    const explicitAccessTokenPath = getCommandLinePathOption(process.argv, "--access-token-path", true);
    const explicitPantalaimonPasswordPath = getCommandLinePathOption(process.argv, "--pantalaimon-password-path", true);
    if (explicitAccessTokenPath !== undefined) {
        config.accessToken = fs.readFileSync(explicitAccessTokenPath, "utf8");
    }
    if (explicitPantalaimonPasswordPath) {
        config.pantalaimon.password = fs.readFileSync(explicitPantalaimonPasswordPath, "utf8");
    }
    return config;
}
```

How about something like this?

Reply:
Guessing the intent is to separate concerns between functions?

Reply:
Really it was to remove the `var` for config (which should have also been a `let`). Another style choice in Draupnir that should be documented is avoiding `let` wherever possible (but there are still reasons to use it sometimes).  But yeah i guess.